comictagger: init at 1.5.5

nodupe opened the pull request:
Adds comictagger, a multi-platform app for writing metadata to digital comics, written in Python and PyQt.

The github project can be found here:
https://github.com/comictagger/comictagger

The following are python dependencies needed:
* py7zr
* pyinstaller
* pyinstaller-hooks-contrib
* settngs
* text2digits
* unrar-cffi
* wordninja

In this commit there are also those packages.

This is my first package, I expect "obvious" mistakes.

I've built and successfully run it on linux, for sure Mac dependencies are different.

Note: To work on wayland it needs to be called from the command line with a specific environemnt variable - I do not know how to handle it at package level:

$ QT_QPA_PLATFORM=wayland comictagger

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Commit:
comictagger: init at 1.5.5

Adds comictagger, a multi-platform app for writing metadata to
digital comics, written in Python and PyQt.

The github project can be found here:
https://github.com/comictagger/comictagger

The following are python dependencies needed:
* py7zr
* pyinstaller
* pyinstaller-hooks-contrib
* settngs
* text2digits
* unrar-cffi
* wordninja

In this commit there are also those packages.

This is my first package, I expect "obvious" mistakes.

I've built and successfully run it on linux, for sure Mac dependencies
are different.

Note: To work on wayland it needs to be called from the command line
with a specific environemnt variable - I do not know how to handle it
at package level:

$ QT_QPA_PLATFORM=wayland comictagger

Commit:
Fixes pointed by github checker

Commit:
Merge branch 'master' of https://github.com/NixOS/nixpkgs into comictagger

Commit:
Removed not needed package and corrected placement of qt5.qtwayland

fixed typos
Moved to pyproject module settngs
Restrict the platform for x64_64-linux

Commit:
Removed not needed package and corrected placement of qt5.qtwayland

Commit:
fixed typo on maintainers file

Commit:
Fixed another typo.

Commit:
Moved to pyproject module settngs

Commit:
Restrict the platform for x64_64-linux

nagy commented on pkgs/development/python-modules/py7zr/default.nix:
```diff
@@ -0,0 +1,55 @@
+    description = "Pure python 7-zip library";
+    homepage = "";
+    license = licenses.lgpl2;
+    maintainers = with maintainers; [  ];
```
All these new python modules would need to have maintainers.

nodupe replied:
I will add myself, and follow the guidance of a separate commit.

nagy commented on pkgs/development/python-modules/py7zr/default.nix:
```diff
@@ -0,0 +1,55 @@
+
+  meta = with lib; {
+    description = "Pure python 7-zip library";
+    homepage = "";
```
A homepage would be nice to find.

nodupe replied:
ok, commit incoming.

nagy commented on pkgs/by-name/co/comictagger/package.nix:
```diff
@@ -0,0 +1,68 @@
+  version = "1.5.5";
+  format = "pyproject";
+
+  src = fetchgit {
```
can you use `fetchFromGitHub` ?

nodupe replied:
Hi, it will not work with fetch from github - it fetches the repo, but later will not compiling, requesting to use the what leaveDotGit = "true"; achieves.

nagy reviewed (changes requested):
Can you follow the commit message conventions at https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions ? We would need one commit for your addition to the maintainers list and one commit for each package, that is being introduced.

Commit:
Added to Maintainer list

Commit:
py7zr: init at 0.20.6

py7zr -- a 7z library on python - py7zr is a library and utility to support 7zip archive compression, decompression, encryption and decryption written by Python programming language.

Commit:
pyinstaller: init at 6.0

PyInstaller bundles a Python application and all its dependencies into a single package. The user can run the packaged app without installing a Python interpreter or any modules.

Commit:
pyinstaller-hooks-contrib: init at 2023.09

collection of hooks for many packages, and allows PyInstaller to work with these packages seamlessly.

Commit:
settngs: init at 0.7.2

This library is an attempt to merge reading flags/options from the commandline (argparse) and settings from a file (json).

Commit:
text2digits: init at 0.10.0

Converts text such as "twenty three" to number/digit "23" in any sentence

Commit:
unrar-cffi: init at 0.2.2

Python unrar cffi edition

Commit:
wordninja: init at 2.0.0

Probabilistically split concatenated words using NLP based on English Wikipedia unigram frequencies.

Commit:
Merge branch 'comictagger' of github.com:nodupe/nixpkgs into comictagger

Commit:
Fixed issues with the repository

Commit:
prettify nix code

Commit:
force linux x86_64

nodupe commented:
I need to work on this before commiting again. All sort of weird issues with my local build.
